### PR TITLE
[Auth] Address possible race condition

### PR DESCRIPTION
**Pull request by paulb777**
This PR eliminates the possibility of multiple AuthBackendRPCImplementation instances other than in test code.

It's a possible fix for https://github.com/firebase/firebase-ios-sdk/issues/13761

(Also eliminates an Objective-C unit test that was relying upon the broken old code to succeed on GHA.

### CODE REVIEW

File: FirebaseAuth/Tests/Unit/ObjCAPITests.m
Comment (ncooke3) [inline FirebaseAuth/Tests/Unit/ObjCAPITests.m:1]: Why the changes in these files?

**Review by ncooke3 — APPROVED** [review]
good idea!

File: FirebaseAuth/Tests/Unit/ObjCAPITests.m
Comment (paulb777) [reply FirebaseAuth/Tests/Unit/ObjCAPITests.m:1]: I was a investigating the test failure. I have a better idea that I'll sort out today.
